lib/core/src/Cardano/Wallet.hs

```diff
@@ -788,6 +803,20 @@ restoreBlocks ctx wid blocks nodeTip = db & \DBLayer{..} -> do
     isParentOf cp = (== parent) . parentHeaderHash . header
       where parent = headerHash $ currentTip cp
 
+-- | Store the node tip params into the wallet database
+saveParams
+    :: forall ctx s k.
+        ( HasLogger WalletLog ctx
+        , HasDBLayer s k ctx
+        )
+    => ctx
+    -> WalletId
+    -> TxParameters
+    -> ExceptT ErrNoSuchWallet IO ()
+saveParams ctx wid params = db & \DBLayer{..} ->
+   mapExceptT atomically $ putTxParameters (PrimaryKey wid) params
```

Comment:
What about trying to get this done in the same atomic block storing the corresponding checkpoint to keep the data consistent in case a failure arises just at the bad time.

Reply:
I don't think it will be necessary to do this because the tip tx parameters and wallets checkpoints are updated independently and asynchronously anyway.